action/logging/pretty_handler.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+
+// Package logging for dealing with logging, log settings and log format
+//
+//	Inspiration:
+//		https://github.com/golang/example/tree/master/slog-handler-guide
+//		https://dusted.codes/creating-a-pretty-console-logger-using-gos-slog-package
+package logging
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"runtime"
+	"sync"
+)
+
+var (
+	errInnerHandle          = errors.New("calling inner handler's Handle failed")
+	errInnerHandleUnmarshal = errors.New("unmarshal output on inner handler's Handle failed")
+)
+
+// PrettyHandler is for prettier logs
+type PrettyHandler struct {
+	baseHandler slog.Handler
+	opts        *slog.HandlerOptions
+	mutex       *sync.Mutex
+	buffer      *bytes.Buffer
+	indent      bool
+	output      io.Writer
+}
+
+// NewPrettyHandler returns a PrettyHandler
+func NewPrettyHandler(output io.Writer, opts *slog.HandlerOptions, indent bool) *PrettyHandler {
+	buffer := &bytes.Buffer{}
+
+	h := &PrettyHandler{
+		baseHandler: slog.NewJSONHandler(
+			buffer,
+			opts,
+		),
+		mutex:  &sync.Mutex{},
+		buffer: buffer,
+		indent: indent,
+		output: output,
+		opts:   opts,
+	}
+	if h.opts.Level == nil {
+		h.opts.Level = slog.LevelInfo
+	}
+	return h
+}
+
+// Enabled reports whether the handler handles records at the given level
+func (h *PrettyHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	return h.baseHandler.Enabled(ctx, level)
+}
+
+// WithAttrs returns a new Handler whose attributes consist of both the receiver's attributes
+// and the arguments
+func (h *PrettyHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	// TODO
+	return &PrettyHandler{
+		baseHandler: h.baseHandler.WithAttrs(attrs),
+		mutex:       h.mutex,
+	}
+}
+
+// WithGroup returns a new Handler with the given group appended to the receiver's existing groups
+func (h *PrettyHandler) WithGroup(name string) slog.Handler {
+	// TODO
+	return &PrettyHandler{
+		baseHandler: h.baseHandler.WithGroup(name),
+		opts:        h.opts,
+		mutex:       h.mutex,
+		output:      h.output,
+	}
+}
+
+// Handle handles the Record
+func (h *PrettyHandler) Handle(ctx context.Context, r slog.Record) error {
+	// Add location where slog.<LEVEL> has been called
+	pc, _, _, _ := runtime.Caller(3)
+	//   0 = logging.(*PrettyHandler).Handle
+	//   1 = slog.(*Logger).log
+	//   2 = slog.Debug || slog.Info || slog.Warn || slog.Error
+	//   3 = place where we call slog.<LEVEL>
+	funcName := TrimNameFunction(pc)
+	r.Add("location", funcName)
```

Comment:
I am not sure if this hard-coded value `runtime.Caller(3)` is OK, but I think it would be difficult to resolve on the fly. I suppose one way would be to iterate over `runtime.Callers`  and find the proper level with regex?

Since it should not change, maybe can stay this way. Also not sure how much overhead dynamically figuring out would cost.

Reply:
There is also `debug.Stack()` but I rarely ever use those methods manually so I'd have too look into the docs for those to figure out a smarter way to get the trace.